src/psyclone/transformations.py

```diff
@@ -1361,3 +1363,83 @@ def apply(self, loop, depth=None):
         loop.update_halo_exchanges()
 
         return schedule, keep
+
+
+class LoopSwapTrans(Transformation):
+    ''' Provides a loop-swap transformation.
+        For example:
+
+        >>> from parse import parse
+        >>> from psyGen import PSyFactory
+        >>> ast,invokeInfo=parse("dynamo.F90")
+        >>> psy=PSyFactory("dynamo0.1").create(invokeInfo)
+        >>> schedule=psy.invokes.get('invoke_v3_kernel_type').schedule
+        >>> schedule.view()
+        >>>
+        >>> from transformations import LoopSwapTrans
+        >>> swap=LoopSwapTrans()
+        >>> new_schedule,memento=swap.apply(schedule.children[0],
+                                             schedule.children[1])
+        >>> new_schedule.view()
+    '''
+
+    def __str__(self):
+        return "Exchange the order of two nested loops: inner becomes " + \
+               "outer and vice versa"
+    @property
+    def name(selfself):
+        '''Returns the name of this transformation as a string'''
+        return "LoopSwap"
+
+    def _validate(self, node_outer):  # pylint: disable=no-self-use
+        ''' validity checks for input arguments '''
+
+        from .psyGen import Loop
+        if not isinstance(node_outer, Loop):
+            raise TransformationError("Error in LoopSwap transformation. "
+                                      "Given node is not a loop.")
```

Comment:
Please add to this message to say what the supplied node actually is to help the user identify where they're going wrong.

Reply:
Not sure what the best way is to show the data, but I did add something - feedback welcome.